cmd/shadertool/main.go

```diff
@@ -43,30 +46,43 @@ func run(ctx context.Context) error {
 	args := flag.Args()
 	if len(args) == 0 {
 		flag.Usage()
+		return nil
 	}
 
 	// Read input
-	input := args[0]
-	source, err := ioutil.ReadFile(input)
-	if err != nil {
-		return err
-	}
-
-	// Process the shader
-	result, err := convert(string(source), filepath.Ext(input))
-	if err != nil {
-		return err
-	}
-
-	// Write output
-	if *output == "" {
-		fmt.Print(result)
-	} else {
-		err := ioutil.WriteFile(*output, []byte(result), 0666)
+	var wg sync.WaitGroup
+	for _, input := range args {
+		input := input
+		source, err := ioutil.ReadFile(input)
 		if err != nil {
 			return err
 		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			// Process the shader
+			result, err := convert(string(source), filepath.Ext(input))
+			if err != nil {
+				fmt.Printf("%v: %v\n", input, err)
+				return
+			}
+
+			// Write output
+			if *out == "" {
+				fmt.Print(result)
```

Comment:
Nit: not great to be printing the output of multiple shaders concurrently, but I guess this is just a debug tool.